uefi/CHANGELOG.md

```diff
@@ -11,20 +11,24 @@
 - `PcrEvent`/`PcrEventInputs` impl `Align`, `Eq`, and `PartialEq`.
 - Added `PcrEvent::new_in_box` and `PcrEventInputs::new_in_box`.
 - `VariableKey` impls `Clone`, `Eq`, `PartialEq`, `Ord`, `PartialOrd`, and `Hash`.
+- The traits `MemoryMap` and `MemoryMapMut` have been introduced together with
+  the implementations `MemoryMapRef`, `MemoryMapRefMut`, and `MemoryMapOwned`.
+  This comes with some changes. Read below.
 
-## Changed
-- **Breaking:** `uefi::helpers::init` no longer takes an argument.
+## Changed (**BREAKING**)
```

Comment:
The `breaking` label is only needed for changes that are semver-incompatible, but not all of these are. I believe that the `SearchType` lifetime change should only allow more code to compile, all existing code should be OK. Looking at previous releases there are other examples: marking something as `deprecated` isn't a breaking change, nor is making a function `const`. So, I'd prefer to keep the `breaking` label on individual entries, even if it is applied to many. (I don't think we need to retroactively do this for all previous releases.)

Reply:
> The breaking label is only needed for changes that are semver-incompatible

To me, `Changed` for a new release is a synonym to `Breaking`. In my mental model, **every** list item in a  `Changed` section is a breaking change, either in (subtle) runtime behavior or build errors after an upgrade.

What are Changes (not under the `Added` section) that are not breaking to you?

Reply:
The distinction I have in mind is this: if an existing user of the crate upgrades to the newer version, what code changes do they have to make _immediately_ just to get the code compiling. In other words, [semver-incompatible](https://doc.rust-lang.org/cargo/reference/semver.html#api-compatibility) changes.

And I define an additive change in the semver sense: things like adding a function, adding a type, adding a member to a `#[non_exaustive]` enum, etc.

Some examples of changes that are not additive and also not breaking:
* Changing a function arg to a less restrictive type (`&[u8]` to `AsRef<[u8]>`)
* Marking an item as deprecated (it would only become a breaking change once the item is fully removed, otherwise it's just a warning)
* Making a function `const` (existing calls will compile fine, it can just be used in more places now)
* Making an `unsafe` function safe (might cause a warning due to no-longer-needed `unsafe` blocks, but will still compile)

Reply:
Got it, sorry. Yes, you are right. Thanks! I'll roll that back.